packages/upload-media/src/index.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Internal dependencies
+ */
+import { store as uploadStore } from './store';
```

Comment:
Hey @swissspidy Didn't we discuss moving the store to the block-editor store instead to avoid the need to create substores? What's the status there?

Reply:
We discussed that this whole client-side media functionality should be added through the block-editor package, but it makes sense to be in its own package (which will be added as a dependency). At least that's how I understood it.

Are you saying you want the whole functionality merged into to the block-editor package instead? I don't think that's a good idea. It's [a lot of code](https://github.com/swissspidy/media-experiments/tree/8b64ed6150aca612cc649f32861c8049e79cd985/packages/upload-media) and would just create a maintenance and reusability mess, with zero separation of concerns.

Reply:
yes, my preference is that it's a sub reducer of the block-editor package as otherwise, we'll have actually more complexity to deal with. For every block editor instance, we'd have to create a block editor store instance and a a media upload store instance. It's doable but for me unnecessary.  Reducers can be combined to create separate concerns in general.

Reply:
Ok, that's a good argument, in that case, I would suggest adding g a `MediaUploadProvider` component (similar to `BlockEditorProvider` that is responsible for initializing the specific store instance and also have support for `useSubRegistry` prop if we want to allow plugins to access this store.

You can see how we do that for the block editor store using this utility (we can just copy it almost) https://github.com/WordPress/gutenberg/blob/0b76a258213bde016726e16949d76fed12c5a34d/packages/block-editor/src/components/provider/with-registry-provider.js

--
Then we'd probably have to render `MediaUploadProvider` within `BlockEditorProvider` (just like we do `BlockEditorProvider` within `EditorProvider` now)

Reply:
OK. I'll try to do that in a separate sub-PR for easier reviewing.

Reply:
Just opened #66380 for this now so to unblock this PR.